### PR TITLE
[KDESKTOP-1258] Prepare the update local node name request before Sync DB upgrade

### PR DESCRIPTION
**Pull request by luc-guyot-infomaniak**

### CODE REVIEW

**Review by ChristopheLarchier — APPROVED** [review]

File: src/libparms/db/parmsdb.cpp [hunk 1]
```diff
@@ -520,6 +520,9 @@ std::shared_ptr<ParmsDb> ParmsDb::instance(const std::filesystem::path &dbPath,
             throw std::runtime_error("ParmsDb must be initialized!");
         } else {
             _instance = std::shared_ptr<ParmsDb>(new ParmsDb(dbPath, version, autoDelete, test));
+            if (!_instance->init(version)) {
+                throw std::runtime_error("ParmsDb initialisation error!");
```
Comment (ChristopheLarchier) [inline src/libparms/db/parmsdb.cpp:524]: Delete _instance in this case?

File: src/libsyncengine/syncpal/syncpal.cpp [hunk 2]
```diff
@@ -633,6 +633,11 @@ bool SyncPal::createOrOpenDb(const SyncPath &syncDbPath, const std::string &vers
     // Create/open sync DB
     try {
         _syncDb = std::shared_ptr<SyncDb>(new SyncDb(syncDbPath.string(), version, targetNodeId));
+        if (!_syncDb->init(version)) {
+            LOG_SYNCPAL_ERROR(_logger, "Database initialisation error");
```
Comment (ChristopheLarchier) [inline src/libsyncengine/syncpal/syncpal.cpp:637]: Idem previous comment

File: src/libparms/db/parmsdb.cpp (hunk 1 above)
Comment (luc-guyot-infomaniak) [reply src/libparms/db/parmsdb.cpp:525]: Done.

File: src/libsyncengine/syncpal/syncpal.cpp (hunk 2 above)
Comment (luc-guyot-infomaniak) [reply src/libsyncengine/syncpal/syncpal.cpp:638]: Done.

**Pull request merged**